### DISCUSSION TITLE
Can't get it working

### DISCUSSION THREAD

**Discussion by pbrink21** [vueform/vueform#126] [Questions] [answered]
I'm having a lot of trouble getting Vueform working. I really want to use it for my application but it's just not working for me for some reason. I can't figure out why... it seems like it should be working from what I understand about the plugin installation tutorials and stuff. 

This is the error: `Uncaught TypeError: can't define property "emits": Object is not extensible`
It's on `index.js line 7`, which is where the routes are declared for the Vue router
I'm just trying to get a basic contact form onto my website 

package.json
```
{
  "name": "als-website",
  "version": "0.0.0",
  "private": true,
  "type": "module",
  "scripts": {
    "dev": "vite",
    "build": "vite build",
    "preview": "vite preview"
  },
  "dependencies": {
    "@fortawesome/fontawesome-svg-core": "^6.5.1",
    "@fortawesome/free-brands-svg-icons": "^6.5.1",
    "@fortawesome/free-regular-svg-icons": "^6.5.1",
    "@fortawesome/free-solid-svg-icons": "^6.5.1",
    "@fortawesome/vue-fontawesome": "^3.0.5",
    "@vueform/vueform": "^1.6.6",
    "bootstrap": "^5.3.2",
    "jquery": "^3.7.1",
    "vue": "^3.4.6",
    "vue-router": "^4.2.5",
    "vue3-carousel": "^0.3.1"
  },
  "devDependencies": {
    "@vitejs/plugin-vue": "^4.5.2",
    "vite": "^5.0.10"
  }
}
```

vite.config.js
```
import { fileURLToPath, URL } from 'node:url'

import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'
import vueform from "@vueform/vueform/vite";

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    vue(),
    vueform(),
  ],
  resolve: {
    alias: {
      '@': fileURLToPath(new URL('./src', import.meta.url))
    }
  }
})
```

vueform.config.js
```
// vueform.config.(js|ts)

import en from '@vueform/vueform/locales/en'
import vueform from '@vueform/vueform/themes/vueform'
import { defineConfig } from '@vueform/vueform'

// You might place these anywhere else in your project
import '@vueform/vueform/themes/vueform/css/index.min.css';

export default defineConfig({
    theme: vueform,
    locales: { en },
    locale: 'en',
})
```
main.js
```
import 'bootstrap/dist/css/bootstrap.css'
import "bootstrap"
import './assets/main.css'

import { createApp } from 'vue'
import App from './App.vue'
import router from './router'
import {Vueform} from "@vueform/vueform";
import vueformConfig from "./../vueform.config.js"; 

//fontawesome stuff

const app = createApp(App)

app.component('font-awesome-icon', FontAwesomeIcon)
app.use(router, Vueform, vueformConfig)

app.mount('#app')
```

app.vue
```
<script setup>
import { RouterLink, RouterView } from 'vue-router'
import {onMounted, ref} from 'vue'
import $ from 'jquery'
import {FontAwesomeIcon} from "@fortawesome/vue-fontawesome";
import FooterComponent from "@/components/FooterComponent.vue";
import NavbarComponent from "@/components/NavbarComponent.vue";
import { Vueform, TextElement } from "@vueform/vueform";

</script>

<template>
  <header id="header-bar" class="d-flex flex-row flex-wrap justify-content-start align-items-start justify-content-lg-center align-items-lg-center ">
    <NavbarComponent ref="navbar"/>
  </header>

  <body>
    <RouterView />
    <Vueform>
      <TextElement name="hello_world" label="Hello" placeholder="World" />
    </Vueform>
  </body>

  <footer>
    <FooterComponent/>
  </footer>
</template>

<style scoped>
//style things
</style>

```

I'm probably just missing something, but I'm kind of at a loss right now. Any guidance would be appreciated. Thanks!

**Comment by adamberecz** [comment]
You don't need the `import { Vueform, TextElement } from "@vueform/vueform";` in `app.vue`.

**Reply by pbrink21** [reply]
Without that, it says that it can't find the elements and doesn't render them. This is the error I get there:
`[Vue warn]: Failed to resolve component: Vueform
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. 
  at <App>`

**Reply by adamberecz** [reply]
What @julisch94 mentioned in another comment changing this:

```js
app.use(router, Vueform, vueformConfig)
```

to this:

```js
app.use(Vueform, vueformConfig)
app.use(router)
```

should solve this issue.

What error you get after that? Can you confirm the error disappears if you remove Vueform from your project? The error message itself does not seem to be familiar to me, which is why I suspect it might come from something else.

**Reply by pbrink21** [reply]
Yeah... unfortunately that didn't fix it either :(
Maybe! It does only happen with Vueform, everything works other than that... but it might be that something is conflicting. I'll have to try some more stuff when I'm at work on Monday

**Reply by pbrink21** [reply]
Doing this just doesn't seem to change anything
Or maybe it did? I did notice this error now too, not sure if that was there before and I just missed it or what:
`[Vue warn]: A plugin must either be a function or an object with an "install" function.`

**Comment by julisch94** [comment]
just guessing here, but can you go into your `main.js` and change this?
```ts
app.use(router, Vueform, vueformConfig)
```
to this:
```ts
app.use(Vueform, vueformConfig)
app.use(router)
```

**Reply by pbrink21** [reply]
Thanks, but no dice :(

**Reply by julisch94** [reply]
since the error seems to be occurring in your `index.js` it might make sense for you to share that file as well :)

**Reply by pbrink21** [reply]
Ahhh yes! Apologies, this is a work project, so I won't have access to the code until Monday. It's a fairly basic one but I'll still post that on Monday. Thank you for helping me with this!

**Reply by pbrink21** [reply]
Here is my index.js:
```

import { createRouter, createWebHistory } from 'vue-router'
import HomeView from '../views/HomeView.vue'
import ContactView from "@/views/ContactView.vue";
import PageNotFound from "@/views/PageNotFound.vue";
import SubmissionSuccess from "@/views/SubmissionSuccess.vue";
import SubmissionFailure from "@/views/SubmissionFailure.vue";

const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes: [
    // Main pages
    { path: '/', name: 'home', component: HomeView },
    { path: '/contact', name: 'contact', component: ContactView },

    // Supporting pages
    { path: '/contact-success', name: 'contact success', component: SubmissionSuccess },
    { path: '/contact-fail', name: 'contact failure', component: SubmissionFailure },

    // 404 page
    { path: '/:pathMatch(.*)*', name: '404', component: PageNotFound }
  ]
})

export default router
```

**Comment by pbrink21** [comment, marked as answer]
I ended up getting it working - it was the curly brackets on this line in main.js:
`import {Vueform} from "@vueform/vueform";`
turning it to
`import Vueform from "@vueform/vueform";`

Apologies for the confusion, thank you all for your time!
